src/atcd/main.go

```diff
@@ -36,8 +36,24 @@ func main() {
 		}
 	}
 
+	// Hooks are called with ATCD_ADDR set to a suitable address
+	// for the atc daemon. For this to work, we need to get the address
+	// and pass it to the engine
+	engine_addr := &net.TCPAddr{
+		Port: args.ThriftAddr.Port,
+		Zone: args.ThriftAddr.Zone,
+	}
+	// zero addresses mean listen to everything, so we use the loopback
+	if args.ThriftAddr.IP.Equal(net.IPv4zero) {
+		engine_addr.IP = net.IPv4(127, 0, 0, 1)
+	} else if args.ThriftAddr.IP.Equal(net.IPv6zero) {
+		engine_addr.IP = net.IPv6loopback
+	} else {
+		engine_addr.IP = args.ThriftAddr.IP
+	}
+
```

Comment:
They're two different piece of information. The ShapingEngine needs an address that it can run http requests against, the thrift server needs an address that it can bind to. The only place both the ShapingEngine and the thrift server are together is this function.
I can abstract this out into a function, but passing the engine a bind address and having it convert it to a dial-able address seems weirder.